pkg/service/query.go

```diff
@@ -22,6 +24,9 @@ func NewQueryService(jaeger *v1alpha1.Jaeger, selector map[string]string) *v1.Se
 			Name:      GetNameForQueryService(jaeger),
 			Namespace: jaeger.Namespace,
 			Labels:    selector,
+			Annotations: map[string]string{
+				"service.alpha.openshift.io/serving-cert-secret-name": GetTLSSecretNameForQueryService(jaeger),
```

Comment:
As mentioned above, it would be good if things like this could be moved into a platform specific SPI.

Reply:
Good catch. I think this will be the only place outside of the `inject` package (and the `normalize`) that we have OpenShift-specific knowledge for the OAuth Proxy (the other feature being Routes vs. Ingresses). For now, I'll just add a check on the OAuth Proxy flag and wait for a second implementation before extracting a provider interface.